coresimd/arm/mod.rs

```diff
@@ -7,14 +7,21 @@
 //! http://infocenter.arm.com/help/topic/com.arm.doc.
 //! ihi0073a/IHI0073A_arm_neon_intrinsics_ref.pdf
 //! [arm_dat]: https://developer.arm.com/technologies/neon/intrinsics
+#![allow(non_camel_case_types)]
 
 mod v6;
 pub use self::v6::*;
 
+#[cfg(any(target_arch = "aarch64", target_feature = "v7"))]
 mod v7;
+#[cfg(any(target_arch = "aarch64", target_feature = "v7"))]
```

Comment:
I'm quite a newbie when it comes to ARM intrinsics, but how come the `#[cfg]` is needed here? 

It seems like `aarch64` implies the `v7` feature (although rustc doesn't print that out?) but is it a bug in rustc/llvm that we can only define the module when the `v7` feature is compiled in?

Reply:
Yes, AArch64 always have a v7 feature enabled, so the LLVM backend does not advertise it as a feature (the v7 feature for AArch64 does not exist).

The cfg checks that either we are targeting for AArch64, or we have v7 enabled (that is, we are targetting ARM, but with v7). 

I don't think this is a bug. There are `arm-unknown...` and `armv7-unknown...` targets. I think this feature can only be enabled at compile time, and maybe linking between binaries compiled w/o this feature is not allowed ? (they might be different targets for a reason?). @japaric ?

Reply:
Oh hm ok so does v7 behave differently from like AVX on x86_64? It's not enabled in the baseline target but we still provide it all the time, right?

Reply:
So for `Aarch64` it is always enabled.

For `arm` AFAICT it must be enabled in the baseline target. That is, `arm-unknown...` does not support it at all, but `armv7-unknown` does. It is also not detectable at run-time AFAICT from looking at the linux hwcap files. So it looks to me that it is an option that `v7` must be enabled statically for the hole binary for `arm` targets. It might well be that `arm-unknown-... + target-feature=v7` is not equal to `armv7-unknown-...` because we are missing some features that are necessary (hence why the `armv7-..` target exists).